ecosystem/plugin-external-link-icon/src/client/config.ts

```diff
@@ -9,7 +10,11 @@ const locales = __EXTERNAL_LINK_ICON_LOCALES__
 
 export default defineClientConfig({
   enhance({ app }) {
-    // wrap the `<ExternalLinkIcon />` component with plugin options
-    app.component('ExternalLinkIcon', h(ExternalLinkIcon, { locales }))
+    // only register when ExternalLinkIcon is not found in global components
+    // so that users can provide their own icon component
+    if (hasGlobalComponent('ExternalLinkIcon', app)) {
+      // wrap the `<ExternalLinkIcon />` component with plugin options
+      app.component('ExternalLinkIcon', h(ExternalLinkIcon, { locales }))
+    }
```

Comment:
This is more likely to be an example why we need `hasGlobalComponent`, if you prefer to have this change in a sperate pr, I can revert it and open another one once you merged this pr.